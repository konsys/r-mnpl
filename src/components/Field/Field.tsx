import React from "react";
export interface FieldProps {
  id: number;
  imgSrc?: string;
  name: string;
  mnplSpecial?: number;
  price?: number;
  mnplGroup?: number;
  mnplCorner?: number;
  mnplLine?: number;
  isJail?: boolean;
}

export const Field = (props: FieldProps) => {
  const {
    mnplCorner,
    mnplLine,
    mnplGroup,
    mnplSpecial,
    isJail,
    price,
    imgSrc
  } = props;

  return (
    <div
      mnpl-corner={mnplCorner}
      mnpl-line={mnplLine}
      mnpl-group={mnplGroup}
      mnpl-special={mnplSpecial}
      className="table-body-board-fields-one"
    >
      {price && (
        <div className="table-body-board-fields-one-label">
          <div>{price}</div>
        </div>
      )}

      <div className="table-body-board-fields-one-body">
        {isJail && (
          <>
            <div className="_jail-visit" />
            <div className="_jail-cell">
              <div />
            </div>
          </>
        )}
        <div className="_logo" style={{ backgroundImage: `url(${imgSrc}` }} />
      </div>
      {!mnplSpecial && <div className="table-body-board-fields-one-level" />}
    </div>
  );
};
