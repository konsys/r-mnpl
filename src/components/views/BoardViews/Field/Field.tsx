import { FieldType, IField, IPlayer } from "../../../../types/types";

import React from "react";
import { createImgPath } from "utils/fields.utils";

interface IProps extends IField {
  players: IPlayer[];
}
export const Field = ({
  fieldCorner,
  fieldLine,
  fieldGroup,
  fieldSpecial,
  isJail,
  price,
  imgSrc,
  status,
  type,
  currency,
  rent,
  fieldPosition,
  onClick,
  players,
}: IProps) => {
  return (
    <>
      <div
        mnpl-corner={fieldCorner}
        mnpl-line={fieldLine}
        mnpl-group={fieldGroup}
        mnpl-special={fieldSpecial}
        mnpl-owner={
          status && players.find((v) => v.userId === status.userId)?.moveOrder
        }
        className={"table-body-board-fields-one"}
        onClick={onClick}
        mnpl-mortgaged={status?.mortgaged && 1}
      >
        {(type === FieldType.AUTO ||
          type === FieldType.COMPANY ||
          type === FieldType.IT) && (
          <div
            mnpl-currency={currency}
            className={`table-body-board-fields-one-label`}
          >
            <div>
              {status?.userId
                ? (status?.branches === 0 &&
                    status?.isMonopoly &&
                    rent?.monopolyRent) ||
                  (status?.branches === 0 && rent?.baseRent) ||
                  (status?.branches === 1 && rent?.oneStar) ||
                  (status?.branches === 2 && rent?.twoStar) ||
                  (status?.branches === 3 && rent?.freeStar) ||
                  (status?.branches === 4 && rent?.fourStar) ||
                  (status?.branches === 5 && rent?.bigStar)
                : price?.startPrice}
            </div>
          </div>
        )}

        <div className="table-body-board-fields-one-body">
          {isJail && (
            <>
              <div className="_jail-visit" />
              <div className="_jail-cell">
                <div />
              </div>
            </>
          )}
          <div
            className="_logo"
            id={`field-${fieldPosition}`}
            style={{ backgroundImage: `url(${createImgPath(imgSrc || "")}` }}
          />
        </div>
        {type === FieldType.COMPANY &&
          status &&
          status?.branches > 0 &&
          status?.branches < 5 && (
            <div className="table-body-board-fields-one-level">
              {new Array(status?.branches).fill(0).map((k, v) => (
                <span className="_small" key={v}></span>
              ))}
            </div>
          )}
        {type === FieldType.COMPANY && status && status?.branches === 5 && (
          <div className="table-body-board-fields-one-level">
            <span className="_big"></span>
          </div>
        )}
        {!fieldSpecial && <div className="table-body-board-fields-one-level" />}
        {status && status.mortgaged > 0 && (
          <div className="table-body-board-fields-one-mortgaged">
            <div className="">
              <span>{status?.mortgaged} </span>
            </div>
          </div>
        )}
      </div>
    </>
  );
};
