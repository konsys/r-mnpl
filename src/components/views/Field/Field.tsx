import React from "react";
import { FieldStatus } from "../../../types/BoardTypes";

export interface BoardField {
  fieldId?: number;
  fieldPosition: number;
  imgSrc?: string;
  name: string;
  mnplSpecial?: number;
  price?: number;
  mnplGroup?: number;
  mnplCorner?: number;
  mnplLine?: number;
  isJail?: boolean;
  status?: FieldStatus;
}

export const Field = (props: BoardField) => {
  const {
    mnplCorner,
    mnplLine,
    mnplGroup,
    mnplSpecial,
    isJail,
    price,
    imgSrc,
    status,
  } = props;

  const field = (
    <div
      mnpl-corner={mnplCorner}
      mnpl-line={mnplLine}
      mnpl-group={mnplGroup}
      mnpl-special={mnplSpecial}
      mnpl-owner={status && status.owner}
      className="table-body-board-fields-one"
    >
      {price && (
        <div className="table-body-board-fields-one-label">
          <div>{props.price}</div>
        </div>
      )}

      <div className="table-body-board-fields-one-body">
        {isJail && (
          <>
            <div className="_jail-visit" />
            <div className="_jail-cell">
              <div />
            </div>
          </>
        )}
        <div className="_logo" style={{ backgroundImage: `url(${imgSrc}` }} />
      </div>
      {!mnplSpecial && <div className="table-body-board-fields-one-level" />}
    </div>
  );

  return <>{field}</>;
};
