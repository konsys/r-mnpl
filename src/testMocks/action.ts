import { ICurrentAction } from "stores/Board/ActionStore";
import {
  BoardAction,
  IncomeMessageType,
  OutcomeMessageType,
} from "types/types";

export const testDoNothingAction: ICurrentAction = {
  actionId: "testActionId",
  event: {
    action: {
      _id: "testID",
      isModal: false,
      userId: 1,
      type: IncomeMessageType.DO_NOTHING,
    },
  },
};

export const testRollDicesModal: BoardAction = {
  _id: "testBoardAction",
  type: IncomeMessageType.INCOME_ROLL_DICES_MODAL,

  userId: 1,
  title: "testTitleRollDices",
  text: "testTextRollDices",
  isModal: true,
};
export const testRollDicesClicked: BoardAction = {
  _id: "testBoardAction",
  type: OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED,
  userId: 1,
  isModal: false,
};
