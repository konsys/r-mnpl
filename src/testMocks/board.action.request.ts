import { IBoardActionRequest, OutcomeMessageType } from "types/types";

export const testBoardActionRequest:IBoardActionRequest = {
    gameId:'testGameId',
    action:OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED
}