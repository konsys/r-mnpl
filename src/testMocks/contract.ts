import { IContract } from "types/types";

export const testContract: IContract = {
  fieldFromPrice: 1000,
  fieldIdsFrom: [1, 2],
  fieldIdsTo: [3],
  fieldToPrice: 1000,
  fromUserId: 1,
  moneyFrom: 1000,
  moneyTo: 1000,
  toUserId: 2,
};
