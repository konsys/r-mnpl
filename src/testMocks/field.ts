import { FieldStatus, FieldType, IField, IFieldAction } from "types/types";

import { FieldActions } from "components/views/BoardViews/Field/FieldActions/FieldAction";

// import {FieldStatus} from ""

export const testField: IField = {
  fieldId: 1,
  fieldPosition: 1,
  name: "testName",
  type: FieldType.COMPANY,
  price: {
    branchPrice: 1000,
    buyoutPrice: 900,
    pledgePrice: 500,
    startPrice: 2000,
  },
  rent: {
    baseRent: 200,
    monopolyRent: 250,
    oneStar: 300,
    twoStar: 1000,
    freeStar: 1500,
    fourStar: 2000,
    bigStar: 3000,
  },
};

export const testOwnedField: IField = {
  status: {
    level: 0,
    branches: 0,
    isMonopoly: false,
    fieldId: 1,
    userId: 1,
    mortgaged: 0,
    fieldActions: [IFieldAction.MORTGAGE],
  },
  fieldId: 1,
  fieldPosition: 1,
  name: "testName",
  type: FieldType.COMPANY,
  price: {
    branchPrice: 1000,
    buyoutPrice: 900,
    pledgePrice: 500,
    startPrice: 2000,
  },
  rent: {
    baseRent: 200,
    monopolyRent: 250,
    oneStar: 300,
    twoStar: 1000,
    freeStar: 1500,
    fourStar: 2000,
    bigStar: 3000,
  },
};

export const testJailField: IField = {
  fieldId: 10,
  fieldPosition: 10,
  name: "jail",
  type: FieldType.JAIL,
  fieldCorner: 2,
  fieldLine: 1,
  isJail: true,
};
