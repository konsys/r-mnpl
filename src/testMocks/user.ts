import { IPlayer, IUser } from "types/types";
import { PlayerRoomStatus, RoomPlayer } from "stores/Game/Rooms/RoomsModel";

export const testPlayer1: IPlayer = {
  frags: "",
  userId: 1,
  money: 15000,
  vip: true,
  registrationType: "none",
  name: "IVAN",
  team: undefined,
  avatar: "testAvatar1",
  createdAt: new Date(),
  updatedAt: new Date(),
  isActive: false,
  isBlocked: true,
  isActing: true,
  gameId: "",
  doublesRolledAsCombo: 0,
  jailed: 0,
  unjailAttempts: 0,
  meanPosition: 0,
  creditPayRound: false,
  creditNextTakeRound: 0,
  score: 0,
  additionalTime: 0,
  timeReduceLevel: 0,
  creditToPay: 0,
  canUseCredit: false,
  moveOrder: 0,
};

export const testPlayer2: IPlayer = {
  frags: "",
  userId: 1,
  money: 15000,
  vip: true,
  registrationType: "none",
  name: "DIMAN",
  team: undefined,
  avatar: "testAvatar2",
  createdAt: new Date(),
  updatedAt: new Date(),
  isActive: false,
  isBlocked: true,
  isActing: false,
  gameId: "",
  doublesRolledAsCombo: 0,
  jailed: 0,
  unjailAttempts: 0,
  meanPosition: 0,
  creditPayRound: false,
  creditNextTakeRound: 0,
  score: 0,
  additionalTime: 0,
  timeReduceLevel: 0,
  creditToPay: 0,
  canUseCredit: false,
  moveOrder: 0,
};

export const testUser: IUser = {
  userId: 1,
  vip: false,
  name: "IVAN",
  isActive: true,
  isBlocked: false,
};
export const test5User: IUser = {
  userId: 5,
  vip: false,
  name: "NIKOLA",
  isActive: true,
  isBlocked: false,
};
export const testAvatarUser: IUser = {
  avatar: "/test/avatar.jpg",
  userId: 2,
  vip: false,
  name: "IVAN",
  isActive: true,
  isBlocked: false,
};
export const testVipUser: IUser = {
  userId: 3,
  vip: true,
  name: "OLEG",
  isActive: true,
  isBlocked: false,
};

export const testRoomUser: RoomPlayer = {
  playerRoomStatus: PlayerRoomStatus.ACITVE,
  userId: 4,
  vip: true,
  name: "MISHA",
  isActive: true,
  isBlocked: false,
};
