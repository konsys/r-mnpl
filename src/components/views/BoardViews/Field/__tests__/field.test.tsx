import { testField, testJailField, testOwnedField } from "testMocks/field";
import { testPlayer1, testPlayer2 } from "testMocks/user";

import { Field } from "../Field";
import React from "react";
import { shallow } from "enzyme";

describe("Field test", () => {
  it("should render", () => {
    expect(
      shallow(<Field {...testField} onClick={() => null} players={[]} />)
    ).toMatchSnapshot();

    expect(
      shallow(
        <Field
          {...testField}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      )
        .find(".table-body-board-fields-one-label > div")
        .text()
    ).toBe(testField.price?.startPrice.toString());
  });

  it("should click", () => {
    const click = jest.fn();
    shallow(<Field {...testField} onClick={click} players={[]} />)
      .find(".table-body-board-fields-one")
      .simulate("click");
    expect(click).toBeCalledTimes(1);
  });

  it("should click", () => {
    const click = jest.fn();
    shallow(
      <Field
        {...testField}
        onClick={click}
        players={[testPlayer1, testPlayer2]}
      />
    )
      .find(".table-body-board-fields-one")
      .simulate("click");
    expect(click).toBeCalledTimes(1);
  });

  it("should render jail field", () => {
    expect(
      shallow(
        <Field
          {...testJailField}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      ).find("._jail-visit")
    ).toHaveLength(1);
    expect(
      shallow(
        <Field
          {...testJailField}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      ).find("._jail-cell")
    ).toHaveLength(1);
  });

  it("should render big star", () => {
    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status, branches: 5 }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      ).find("._big")
    ).toHaveLength(1);

    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status, branches: 5 }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      )
        .find(".table-body-board-fields-one-label > div")
        .text()
    ).toBe(testOwnedField.rent?.bigStar.toString());
  });
  it("should render 1 star", () => {
    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status, branches: 1 }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      ).find("._small")
    ).toHaveLength(1);
    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status, branches: 1 }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      )
        .find(".table-body-board-fields-one-label > div")
        .text()
    ).toBe(testOwnedField.rent?.oneStar.toString());
  });
  it("should render 2 stars", () => {
    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status, branches: 2 }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      ).find("._small")
    ).toHaveLength(2);

    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status, branches: 2 }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      )
        .find(".table-body-board-fields-one-label > div")
        .text()
    ).toBe(testOwnedField.rent?.twoStar.toString());
  });
  it("should render 3 stars", () => {
    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status, branches: 3 }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      ).find("._small")
    ).toHaveLength(3);
    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status, branches: 3 }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      )
        .find(".table-body-board-fields-one-label > div")
        .text()
    ).toBe(testOwnedField.rent?.freeStar.toString());
  });
  it("should render 4 stars", () => {
    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status, branches: 4 }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      ).find("._small")
    ).toHaveLength(4);
    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status, branches: 4 }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      )
        .find(".table-body-board-fields-one-label > div")
        .text()
    ).toBe(testOwnedField.rent?.fourStar.toString());
  });
  it("should render mortgaged field", () => {
    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status, mortgaged: 1 }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      ).find(".table-body-board-fields-one-mortgaged")
    ).toHaveLength(1);
  });
  it("should render proper field title", () => {
    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status, mortgaged: 1 }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      ).find(".table-body-board-fields-one-label > div")
    ).toHaveLength(1);
  });
  it("should render proper field title", () => {
    expect(
      shallow(
        <Field
          {...testOwnedField}
          status={{ ...testOwnedField.status }}
          onClick={() => null}
          players={[testPlayer1, testPlayer2]}
        />
      )
        .find(".table-body-board-fields-one-label > div")
        .text()
    ).toBe(testOwnedField.rent?.baseRent.toString());
  });
});
// table-body-board-fields-one-label
