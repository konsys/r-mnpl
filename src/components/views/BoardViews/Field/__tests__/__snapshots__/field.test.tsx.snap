// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field test should render 1`] = `ShallowWrapper {}`;
